lib/widgets/message_list.dart

```diff
@@ -849,61 +852,86 @@ class MessageWithPossibleSender extends StatelessWidget {
+                          ).merge(weightVariableTextStyle(context, wght: 600,
+                              wghtIfPlatformRequestsBold: 900)),
+                          overflow: TextOverflow.ellipsis)),
+                    ]))),
+              Text(
+                _kMessageTimestampFormat.format(
+                  DateTime.fromMillisecondsSinceEpoch(1000 * message.timestamp)),
+                style: TextStyle(
+                  color: _kMessageTimestampColor,
+                  fontFamily: 'Source Sans 3',
+                  fontSize: 16,
+                  height: (18 / 16),
+                  fontFeatures: const [FontFeature.enable('c2sc'), FontFeature.enable('smcp')],
+                ).merge(weightVariableTextStyle(context))),
+            ]))
+      : null;
+
+    return GestureDetector(
+      behavior: HitTestBehavior.translucent,
+      onLongPress: () => showMessageActionSheet(context: context, message: message),
+      child: Column(
+        children: [
+          if (senderRow != null) senderRow,
+          Padding(
+            padding: const EdgeInsets.symmetric(vertical: 4),
+            child: Row(
+              crossAxisAlignment: CrossAxisAlignment.start,
               children: [
-                if (item.showSender) ...[
-                  const SizedBox(height: 3),
-                  GestureDetector(
-                    onTap: () => Navigator.push(context,
-                      ProfilePage.buildRoute(context: context,
-                        userId: message.senderId)),
-                    child: Text(message.senderFullName, // TODO get from user data
-                      style: const TextStyle(fontWeight: FontWeight.bold))),
-                  const SizedBox(height: 4),
-                ],
-                MessageContent(message: message, content: item.content),
-                if ((message.reactions?.total ?? 0) > 0)
-                  ReactionChipsList(messageId: message.id, reactions: message.reactions!)
+                const SizedBox(width: 16),
+                Expanded(
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.stretch,
+                    children: [
+                      MessageContent(message: message, content: item.content),
+                      if ((message.reactions?.total ?? 0) > 0)
+                        ReactionChipsList(messageId: message.id, reactions: message.reactions!)
+                    ])),
+                SizedBox(width: 16,
+                  child: message.flags.contains(MessageFlag.starred)
+                    // TODO(#157): fix how star marker aligns with message content
+                    // Design from Figma at:
+                    //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=813%3A28817&mode=dev .
+                    ? Padding(padding: const EdgeInsets.only(top: 4),
+                        child: FittedBox(child: Icon(ZulipIcons.star, color: _kStarColor)))
```

Comment:
In addition to the issue mentioned at the top of the PR description, about headings in the message, it seems like the star is too high even when the message starts with a paragraph.

In your screenshot:
![image](https://github.com/zulip/zulip-flutter/assets/28173/8cdc8663-5ff1-4922-965d-41560dfd4c90)
basically the whole top lobe of the star sticks out above the ascenders of the text. But in the image you quoted at https://github.com/zulip/zulip-flutter/pull/449#discussion_r1446403566 from Vlad's design, the upper tip of the star is pretty much right at the ascender line.

So that would be good to fix.

Reply:
Made a side by side, left is the app and right is the mockup:

![side_by_side](https://github.com/zulip/zulip-flutter/assets/98299/0b38e1fa-6f72-4101-9bfb-b56154ab7bf2)

It looks to me that the star is vertically aligned with the text (there's the same-ish space above and below the `h`) and that's the approach I took to aligning the star with the paragraph.

In the process of making the screenshot I did end up moving the star up 1px so the padding is now `top: 3px` (I was tweaking the alignment based on the numeral `2` but it seems the `h` is a little bit taller). It's not perfectly centered but it's much closer than the previous `top: 4px`.

Reply:
I think one other reason the star looks to ascend even higher is that our unstyled test is currently `14px` (see `kbaseFontSize` as well as `Paragraph` in `lib/widgets/contents.dart`) whereas the mockup has `17px`. This change is incoming - I noticed this during my work on #157. There's already a comment in the code noting that we'll need to revisit this alignment when the content styles are updated.

Reply:
Thanks. Playing with it, the star still feels slighly misaligned to me, and I think a 4px offset (rather than 3px) comes out better.

I think the key to how it looks for me is the shoulders of the star, i.e. the upper sides of the left and right tines — in the current revision those are noticeably higher than the x-height of the text (which is the top of most of the letters), whereas in the mockup it's right in line.  Moving it another pixel down fixes that.